generator/src/lib.rs

```diff
@@ -2474,8 +2548,19 @@ pub fn write_source_code<P: AsRef<Path>>(vk_headers_dir: &Path, src_dir: P) {
     let definition_code: Vec<_> = definitions
         .into_iter()
         .filter_map(|def| {
+            let def2 = definitions2
+                .iter()
+                .filter_map(get_variant!(vk_parse::TypesChild::Type))
+                .find(|def2| match def {
+                    vkxml::DefinitionsElement::Struct(s) => {
+                        def2.name.as_deref() == Some(s.name.as_str())
+                    }
+                    // TODO: Compare the name for more vkxml types
+                    _ => false,
+                });
```

Comment:
This bit is pretty messy but we need the `vk_parse` type as `vkxml` doesn't propagate the `values` field, which tells for every `s_type` what value it *should* be when filled in on a particular type.

Reply:
Resolved this by getting hold of the variant name for `s_type` just how our `Default` implementation already did. Much simpler.